src/index.js

```diff
@@ -1,17 +1,20 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { BrowserRouter, Route } from 'react-router-dom';
+import CounterContainer from './components/CounterPage/CounterContainer';
+import MainPage from './components/MainPage';
 import './index.css';
-import App from './App';
 import reportWebVitals from './reportWebVitals';
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <BrowserRouter>
+      <Route exact path='/' component={MainPage} />
+      <Route exact path='/counter' component={CounterContainer} />
```

Comment:
Create separated component with name Routes and put your Routes there. Don't forget to wrap it in `Switch` component. Also create a file routeNames in which you will store values which you pass to `path` attribute.
And replace hardcoded passes with variables from your routeNames file.